In [1]:
#importing needed libiraries
import pandas as pd

In [2]:
#loading our data
row_data = pd.read_csv("Absenteeism_data.csv")

In [3]:
#making a copy to work with so we can get back to it, in case the copy was spoiled
df = row_data.copy()

In [4]:
#displaying first rows of our dataet 
df.head()

ID  Reason for Absence        Date  Transportation Expense  \
0  11                  26  07/07/2015                     289   
1  36                   0  14/07/2015                     118   
2   3                  23  15/07/2015                     179   
3   7                   7  16/07/2015                     279   
4  11                  23  23/07/2015                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  
0         2     1                          4  
1         1     0                          0  
2         0     0                          2  
3         2     0                          4  
4         2     1                          2

In [5]:
df.shape

(700, 12)

>- num rows = 700
>- num columns = 11

In [6]:
# some information regarding the df
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   ID                         700 non-null    int64  
 1   Reason for Absence         700 non-null    int64  
 2   Date                       700 non-null    object 
 3   Transportation Expense     700 non-null    int64  
 4   Distance to Work           700 non-null    int64  
 5   Age                        700 non-null    int64  
 6   Daily Work Load Average    700 non-null    float64
 7   Body Mass Index            700 non-null    int64  
 8   Education                  700 non-null    int64  
 9   Children                   700 non-null    int64  
 10  Pets                       700 non-null    int64  
 11  Absenteeism Time in Hours  700 non-null    int64  
dtypes: float64(1), int64(10), object(1)
memory usage: 65.8+ KB


In [7]:
df["Date"].describe()

count            700
unique           432
top       17/08/2015
freq               5
Name: Date, dtype: object

> Conclusion:
> - no messing values 
> - date is represented a an opject

> despite that everything looks good so far

In [8]:
#investigating the df statistics
df.describe()

ID  Reason for Absence  Transportation Expense  \
count  700.000000          700.000000              700.000000   
mean    17.951429           19.411429              222.347143   
std     11.028144            8.356292               66.312960   
min      1.000000            0.000000              118.000000   
25%      9.000000           13.000000              179.000000   
50%     18.000000           23.000000              225.000000   
75%     28.000000           27.000000              260.000000   
max     36.000000           28.000000              388.000000   

       Distance to Work         Age  Daily Work Load Average  Body Mass Index  \
count        700.000000  700.000000               700.000000       700.000000   
mean          29.892857   36.417143               271.801774        26.737143   
std           14.804446    6.379083                40.021804         4.254701   
min            5.000000   27.000000               205.917000        19.000000   
25%           16.000000   31.000000               241.476000        24.000000   
50%           26.000000   37.000000               264.249000        25.000000   
75%           50.000000   40.000000               294.217000        31.000000   
max           52.000000   58.000000               378.884000        38.000000   

        Education    Children        Pets  Absenteeism Time in Hours  
count  700.000000  700.000000  700.000000                 700.000000  
mean     1.282857    1.021429    0.687143                   6.761429  
std      0.668090    1.112215    1.166095                  12.670082  
min      1.000000    0.000000    0.000000                   0.000000  
25%      1.000000    0.000000    0.000000                   2.000000  
50%      1.000000    1.000000    0.000000                   3.000000  
75%      1.000000    2.000000    1.000000                   8.000000  
max      4.000000    4.000000    8.000000                 120.000000

## working with id column

In [9]:
#dropping id column as it wouldn't help us with our analysis
df = df.drop(["ID"], axis=1)

In [10]:
#making ID column dropped 
df

Reason for Absence        Date  Transportation Expense  Distance to Work  \
0                    26  07/07/2015                     289                36   
1                     0  14/07/2015                     118                13   
2                    23  15/07/2015                     179                51   
3                     7  16/07/2015                     279                 5   
4                    23  23/07/2015                     289                36   
..                  ...         ...                     ...               ...   
695                  10  23/05/2018                     179                22   
696                   6  23/05/2018                     225                26   
697                  10  24/05/2018                     330                16   
698                  23  24/05/2018                     235                16   
699                  28  31/05/2018                     291                31   

     Age  Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0     33                  239.554               30          1         2     1   
1     50                  239.554               31          1         1     0   
2     38                  239.554               31          1         0     0   
3     39                  239.554               24          1         2     0   
4     33                  239.554               30          1         2     1   
..   ...                      ...              ...        ...       ...   ...   
695   40                  237.656               22          2         2     0   
696   28                  237.656               24          1         1     2   
697   28                  237.656               25          2         0     0   
698   32                  237.656               25          3         0     0   
699   40                  237.656               25          1         1     1   

     Absenteeism Time in Hours  
0                            4  
1                            0  
2                            2  
3                            4  
4                            2  
..                         ...  
695                          8  
696                          3  
697                          8  
698                          2  
699                          2  

[700 rows x 11 columns]

## working with "Reason for Absence" column

##### some information regerding the reson for absence column:
- these numbers are a catigorical nominal data
- each number resfers to an absance reason inserted in this bdf " add a link to the photo in your gethub repo"

such a great way for inserting catigorical data, it'll help us performing regression, predicting the future absence reasons and maybe reducing them ! 


In [11]:
#column deacription
df["Reason for Absence"].describe()

count    700.000000
mean      19.411429
std        8.356292
min        0.000000
25%       13.000000
50%       23.000000
75%       27.000000
max       28.000000
Name: Reason for Absence, dtype: float64

>- this tells us that we have 29 values each value refers to a specific reason for absence
>- min = 0
>- max = 28

In [12]:
#getting the column's values
reasons_vals  = df["Reason for Absence"].unique()
reasons_vals

array([26,  0, 23,  7, 22, 19,  1, 11, 14, 21, 10, 13, 28, 18, 25, 24,  6,
       27, 17,  8, 12,  5,  9, 15,  4,  3,  2, 16], dtype=int64)

#### does all the 29 absence resons occures in our dataset ?

In [13]:
#getting the len of our uniqe values 
len(reasons_vals)

28

#### looks like 1 of the 29 reasons dosen't occure in this dataset, but which one is it?

In [14]:
# a for loop that chicks for every number from 0 to 28
missing_vals = []
for n in range(29):
    if n not in sorted(reasons_vals):
            missing_vals.append(n)

In [15]:
missing_vals

[20]

><p>reason 20 wasn't mentioned in our dataset</p>
><p>maybe we shouldn't consider it as an absence reasone in the next >time, so we can reducing our scope of research</p>

##### now we need to prepare this column to the regression phase by creating dummy variables that represents each num 
##### dummy variables:
an explanatory binary variable that equals :
- 1 if a certIain categorical effect is present. 
- 0 if that same effect is absent.

In [16]:
#creating dummy variables and droping the first column
reason_columns = pd.get_dummies(df["Reason for Absence"], drop_first = True)

In [17]:
reason_columns

1   2   3   4   5   6   7   8   9   10  ...  18  19  21  22  23  24  25  \
0     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
1     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   
2     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
3     0   0   0   0   0   0   1   0   0   0  ...   0   0   0   0   0   0   0   
4     0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
..   ..  ..  ..  ..  ..  ..  ..  ..  ..  ..  ...  ..  ..  ..  ..  ..  ..  ..   
695   0   0   0   0   0   0   0   0   0   1  ...   0   0   0   0   0   0   0   
696   0   0   0   0   0   1   0   0   0   0  ...   0   0   0   0   0   0   0   
697   0   0   0   0   0   0   0   0   0   1  ...   0   0   0   0   0   0   0   
698   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   1   0   0   
699   0   0   0   0   0   0   0   0   0   0  ...   0   0   0   0   0   0   0   

     26  27  28  
0     1   0   0  
1     0   0   0  
2     0   0   0  
3     0   0   0  
4     0   0   0  
..   ..  ..  ..  
695   0   0   0  
696   0   0   0  
697   0   0   0  
698   0   0   0  
699   0   0   1  

[700 rows x 27 columns]

### Group the Reason for absence 

###### discription
- <b>step1</b>: dropping the reasons for absence column to prevent multicollinearity
- <b>step2</b>: gruping the varibles or doing classificion to minimize the number of options we are analysing, so our regression can be more efficint
<ol>
  <li>1-14 are related to various diseases</li>
  <li>15-17 are related to pregnancy and giving birth</li>
  <li>18-21 poisoning</li>
   <li>22-28 light reasons for absence</li>
</ol> 
different grouping methods applies to different cases. you can chick the groups via this link """
- <b>step3</b>: adding the dummy variables to our df

### step 1

In [18]:
#dropping the Reason for Absence column
df = df.drop(["Reason for Absence"], axis=1)

In [19]:
df.head()

Date  Transportation Expense  Distance to Work  Age  \
0  07/07/2015                     289                36   33   
1  14/07/2015                     118                13   50   
2  15/07/2015                     179                51   38   
3  16/07/2015                     279                 5   39   
4  23/07/2015                     289                36   33   

   Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                  239.554               30          1         2     1   
1                  239.554               31          1         1     0   
2                  239.554               31          1         0     0   
3                  239.554               24          1         2     0   
4                  239.554               30          1         2     1   

   Absenteeism Time in Hours  
0                          4  
1                          0  
2                          2  
3                          4  
4                          2

### step 2

<p>in statistics dummy variables should always sum up to 1 across each row each row have only one absence reason.</p>

now we are going to sum up all the dummy values a cross each row within the 4 groups, so that:
>- if we get 1 it means that there is a reason occurs from this specific group
>- if we get 0 that means that the reason for absence is from the other groups

In [20]:
#grouping our dummy variables in reason_columns
#1:14 are related to vaiase diseases
#15:17 are related to pregnancy and giving birth
#18:21 poisoning
#22:28 light reasons for absance
reason_type_1 = reason_columns.loc[:, 1:14].sum(axis = 1)
reason_type_2 = reason_columns.loc[:, 15:17].sum(axis = 1)
reason_type_3 = reason_columns.loc[:, 18:21].sum(axis = 1)
reason_type_4 = reason_columns.loc[:, 22:].sum(axis = 1)

In [21]:
#a sample of the result
reason_type_1

0      0
1      0
2      0
3      1
4      0
      ..
695    1
696    1
697    1
698    0
699    0
Length: 700, dtype: int64

### step 3

In [22]:
# adding the reason columns
df = pd.concat([df, reason_type_1, reason_type_2, reason_type_3, reason_type_4], axis = 1)

In [23]:
df

Date  Transportation Expense  Distance to Work  Age  \
0    07/07/2015                     289                36   33   
1    14/07/2015                     118                13   50   
2    15/07/2015                     179                51   38   
3    16/07/2015                     279                 5   39   
4    23/07/2015                     289                36   33   
..          ...                     ...               ...  ...   
695  23/05/2018                     179                22   40   
696  23/05/2018                     225                26   28   
697  24/05/2018                     330                16   28   
698  24/05/2018                     235                16   32   
699  31/05/2018                     291                31   40   

     Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                    239.554               30          1         2     1   
1                    239.554               31          1         1     0   
2                    239.554               31          1         0     0   
3                    239.554               24          1         2     0   
4                    239.554               30          1         2     1   
..                       ...              ...        ...       ...   ...   
695                  237.656               22          2         2     0   
696                  237.656               24          1         1     2   
697                  237.656               25          2         0     0   
698                  237.656               25          3         0     0   
699                  237.656               25          1         1     1   

     Absenteeism Time in Hours  0  1  2  3  
0                            4  0  0  0  1  
1                            0  0  0  0  0  
2                            2  0  0  0  1  
3                            4  1  0  0  0  
4                            2  0  0  0  1  
..                         ... .. .. .. ..  
695                          8  1  0  0  0  
696                          3  1  0  0  0  
697                          8  1  0  0  0  
698                          2  0  0  0  1  
699                          2  0  0  0  1  

[700 rows x 14 columns]

> <p>looks like the columns was added as 0, 1, 2, 3</p>
> <p>i'm changing them so they are more dscriptive</p>

In [24]:
#first getting the columns names
df.columns.values

array(['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 0, 1, 2, 3],
      dtype=object)

In [25]:
#creating the new columns names
column_names = ['Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 'reason_1', 'reason_2', 'reason_3', 'reason_4']

In [26]:
#assigning the columns names to our df
df.columns = column_names

In [27]:
#chicking our changes
df.head()

Date  Transportation Expense  Distance to Work  Age  \
0  07/07/2015                     289                36   33   
1  14/07/2015                     118                13   50   
2  15/07/2015                     179                51   38   
3  16/07/2015                     279                 5   39   
4  23/07/2015                     289                36   33   

   Daily Work Load Average  Body Mass Index  Education  Children  Pets  \
0                  239.554               30          1         2     1   
1                  239.554               31          1         1     0   
2                  239.554               31          1         0     0   
3                  239.554               24          1         2     0   
4                  239.554               30          1         2     1   

   Absenteeism Time in Hours  reason_1  reason_2  reason_3  reason_4  
0                          4         0         0         0         1  
1                          0         0         0         0         0  
2                          2         0         0         0         1  
3                          4         1         0         0         0  
4                          2         0         0         0         1

In [28]:
#reording columns
#creating a list of the reordered names
columns_names_reordered = [ 'reason_1', 'reason_2', 'reason_3', 'reason_4', 'Date', 'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours']

In [29]:
#assigning the new order
df = df[columns_names_reordered]

In [30]:
#chocking the changes
df.head()

reason_1  reason_2  reason_3  reason_4        Date  Transportation Expense  \
0         0         0         0         1  07/07/2015                     289   
1         0         0         0         0  14/07/2015                     118   
2         0         0         0         1  15/07/2015                     179   
3         1         0         0         0  16/07/2015                     279   
4         0         0         0         1  23/07/2015                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  
0         2     1                          4  
1         1     0                          0  
2         0     0                          2  
3         2     0                          4  
4         2     1                          2

### creating a checkpoint

In [31]:
#creating a checkpoint to tell us what stage of data preprocessing we're at
df_reason_mod = df.copy()

## working with date column

In [32]:
#taking a look at the date column
df_reason_mod["Date"][0]

'07/07/2015'

In [33]:
#cheking the datatpe of date
type(df_reason_mod["Date"][0])

str

In [34]:
#converting the date into datetime
df_reason_mod["Date"] = pd.to_datetime(df_reason_mod["Date"], format='%d/%m/%Y')

In [35]:
#cheking changes
type(df_reason_mod['Date'])

pandas.core.series.Series

In [36]:
df_reason_mod.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 700 entries, 0 to 699
Data columns (total 14 columns):
 #   Column                     Non-Null Count  Dtype         
---  ------                     --------------  -----         
 0   reason_1                   700 non-null    int64         
 1   reason_2                   700 non-null    int64         
 2   reason_3                   700 non-null    int64         
 3   reason_4                   700 non-null    int64         
 4   Date                       700 non-null    datetime64[ns]
 5   Transportation Expense     700 non-null    int64         
 6   Distance to Work           700 non-null    int64         
 7   Age                        700 non-null    int64         
 8   Daily Work Load Average    700 non-null    float64       
 9   Body Mass Index            700 non-null    int64         
 10  Education                  700 non-null    int64         
 11  Children                   700 non-null    int64         
 12  Pets    

### extracting the month value

>we need to check whether a specific month in a year employees tend to be absent more often compared to other months, so we're creating a months column.

In [37]:
df_reason_mod['Date'][0].month

7

In [38]:
list_months = []
for i in range(df_reason_mod.shape[0]):
    list_months.append(df_reason_mod['Date'][i].month)

In [39]:
list_months

[7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 7,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 8,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 9,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 10,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 11,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 12,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 1,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 2,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 3,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 4,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 5,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 6,
 7,
 

In [40]:
len(list_months)

700

In [41]:
df_reason_mod["Month"] = list_months

In [42]:
df_reason_mod.head()

reason_1  reason_2  reason_3  reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  Month  
0         2     1                          4      7  
1         1     0                          0      7  
2         0     0                          2      7  
3         2     0                          4      7  
4         2     1                          2      7

In [43]:
df_reason_mod["Month"].describe()

count    700.00000
mean       6.36000
std        3.50501
min        1.00000
25%        3.00000
50%        6.00000
75%       10.00000
max       12.00000
Name: Month, dtype: float64

### extracting the day value 

> extracting the days for the sake of the same reason mentioned above

In [44]:
#a function to extract days
def weekdays(data):
    return data.weekday()

In [45]:
df_reason_mod["Day of the Week"] = df_reason_mod['Date'].apply(weekdays)

In [46]:
#cheking our work 
df_reason_mod.head()

reason_1  reason_2  reason_3  reason_4       Date  Transportation Expense  \
0         0         0         0         1 2015-07-07                     289   
1         0         0         0         0 2015-07-14                     118   
2         0         0         0         1 2015-07-15                     179   
3         1         0         0         0 2015-07-16                     279   
4         0         0         0         1 2015-07-23                     289   

   Distance to Work  Age  Daily Work Load Average  Body Mass Index  Education  \
0                36   33                  239.554               30          1   
1                13   50                  239.554               31          1   
2                51   38                  239.554               31          1   
3                 5   39                  239.554               24          1   
4                36   33                  239.554               30          1   

   Children  Pets  Absenteeism Time in Hours  Month  Day of the Week  
0         2     1                          4      7                1  
1         1     0                          0      7                1  
2         0     0                          2      7                2  
3         2     0                          4      7                3  
4         2     1                          2      7                3

In [47]:
df_reason_mod.columns.values

array(['reason_1', 'reason_2', 'reason_3', 'reason_4', 'Date',
       'Transportation Expense', 'Distance to Work', 'Age',
       'Daily Work Load Average', 'Body Mass Index', 'Education',
       'Children', 'Pets', 'Absenteeism Time in Hours', 'Month',
       'Day of the Week'], dtype=object)

In [48]:
column_names_upd = ['reason_1', 'reason_2', 'reason_3', 'reason_4','Date', 'Month', 'Day of the Week',
                                'Transportation Expense', 'Distance to Work', 'Age',
                                'Daily Work Load Average', 'Body Mass Index', 'Education', 'Children',
                                'Pets', 'Absenteeism Time in Hours']
df_reason_mod = df_reason_mod[column_names_upd]

## Working with Education column

In [49]:
#investigating the education column
df_reason_mod["Education"].unique()

array([1, 3, 2, 4], dtype=int64)

><b>description:</b><br>
1 = high school<br>
2 = graduate<br>
3 = postgraduate<br>
4 = a master or a doctor<br>

In [50]:
df_reason_mod["Education"].value_counts()

1    583
3     73
2     40
4      4
Name: Education, dtype: int64

> looks like most of employees, about 583 have high school education only,
and all other education type falls in the rest 117.<br>
we can consider them as a one group for the sake of simpliesty.

In [51]:
#creating the group
df_reason_mod["Education"] = df_reason_mod["Education"].map({1:0, 2:1, 3:1, 4:1})

In [52]:
df_reason_mod["Education"].unique()

array([0, 1], dtype=int64)

In [53]:
df_reason_mod["Education"].value_counts()

0    583
1    117
Name: Education, dtype: int64

> 0 = high school<br>
1 = graduate, postgraduate, a master or a doctor

## rest of the columns

- Children and Pets columns are quantitative values so we're not modifying them
- Transportation Expense = Expense expressed by dollars 
- Distance to Work = Distance in kilometre
- Age = a quantitative value
- Daily Work Load Average = expressed in minutes
- Body Mass Index = expressed in kilograms

## Final Chickpoint

In [54]:
df_preprocessed_1 = df_reason_mod.copy()

In [55]:
df_reason_mod.head()

reason_1  reason_2  reason_3  reason_4       Date  Month  Day of the Week  \
0         0         0         0         1 2015-07-07      7                1   
1         0         0         0         0 2015-07-14      7                1   
2         0         0         0         1 2015-07-15      7                2   
3         1         0         0         0 2015-07-16      7                3   
4         0         0         0         1 2015-07-23      7                3   

   Transportation Expense  Distance to Work  Age  Daily Work Load Average  \
0                     289                36   33                  239.554   
1                     118                13   50                  239.554   
2                     179                51   38                  239.554   
3                     279                 5   39                  239.554   
4                     289                36   33                  239.554   

   Body Mass Index  Education  Children  Pets  Absenteeism Time in Hours  
0               30          0         2     1                          4  
1               31          0         1     0                          0  
2               31          0         0     0                          2  
3               24          0         2     0                          4  
4               30          0         2     1                          2

In [56]:
df_preprocessed_1.to_csv('df_preprocessed_2.csv', index=False)